skimage/filters/thresholding.py

```diff
@@ -118,10 +122,22 @@ def try_all_threshold(image, figsize=(8, 5), verbose=True):
     """
     def thresh(func):
         """
-        A wrapper function to return a thresholded image.
+        Select a wrapper function to return a thresholded image. 
         """
-        def wrapper(im):
-            return im > func(im)
+        if ( func.__name__ == 'threshold_local' ):
+            # threshold_local requires the additional parameter 'block_size'
+            def wrapper(im):
+                return im > func(im, 15)
+        elif ( func.__name__ == 'threshold_phansalkar' ):
+            # threshold_phansalkar requires an equalized image
+            def wrapper(im):
+                im_eq = equalize_adapthist(im)
+                return im_eq > func(im_eq)
+        else:
+            # standard case
+            def wrapper(im):
+                return im > func(im)
+
```

Comment:
I don't know if you would accept this as good style, but it works for thresholding algorithms which require different input variables.